src/test/java/hudson/plugins/warnings/parser/IarParserTest.java

```diff
@@ -20,55 +20,146 @@
     private static final String TYPE = new IarParser().getGroup();
 
     /**
-     * Parses a file with warnings that are prefixed with exec tag.
+     * Parses a file with warnings/errors in all styles. it check the amount of error/warnings found
      *
      * @throws IOException
      *      if the file could not be read
      * @see <a href="http://issues.jenkins-ci.org/browse/JENKINS-8823">Issue 8823</a>
      */
     @Test
-    public void issue8823() throws IOException {
+    public void IAR_error_size() throws IOException {
         Collection<FileAnnotation> warnings = new IarParser().parse(openFile("issue8823.txt"));
 
-        assertEquals(WRONG_NUMBER_OF_WARNINGS_DETECTED, 3, warnings.size());
-        FileAnnotation annotation = warnings.iterator().next();
-        checkWarning(annotation, 3767, "enumerated type mixed with another type",
-                "D:/continuousIntegration/modifiedcomps/forcedproduct/MHSM-Cascade/Cascade-Config/config/src/RDR_Config.c",
-                "Pe188", Priority.NORMAL);
+        assertEquals(WRONG_NUMBER_OF_WARNINGS_DETECTED, 6, warnings.size());
     }
 
     /**
-     * Parses a file with two IAR warnings.
+     * Parses a file and check error number 1
      *
      * @throws IOException
      *      if the file could not be read
+     * @see <a href="http://issues.jenkins-ci.org/browse/JENKINS-8823">Issue 8823</a>
      */
-    @Test
-    public void testWarningsParser() throws IOException {
-        Collection<FileAnnotation> warnings = new IarParser().parse(openFile());
-
+   @Test
+    public void IAR_error1() throws IOException {
+        Collection<FileAnnotation> warnings = new IarParser().parse(openFile("issue8823.txt"));
         Iterator<FileAnnotation> iterator = warnings.iterator();
 
-        assertEquals("Wrong number of warnings detected.", 9, warnings.size());
-        checkWarning(iterator.next(), 3, "an inline function cannot be root as well",
-                "/tmp/x/icc-error-memory.c", TYPE, "Be031", Priority.HIGH);
+        FileAnnotation annotation = iterator.next();
+        
+       checkWarning(annotation, 3767, "enumerated type mixed with another type",
+                "D:/continuousIntegration/modifiedcomps/forcedproduct/MHSM-Cascade/Cascade-Config/config/src/RDR_Config.c",
+                "Pe188", Priority.NORMAL);
     }
-
+    
+     /**
+     * Parses a file and check error number 2
+     *
+     * @throws IOException
+     *      if the file could not be read
+     * @see <a href="http://issues.jenkins-ci.org/browse/JENKINS-8823">Issue 8823</a>
+     */
+   @Test
+    public void IAR_error2() throws IOException {
+        Collection<FileAnnotation> warnings = new IarParser().parse(openFile("issue8823.txt"));
```

Comment:
Can you wrap that into a method with an integer parameter for the warning number to return?
```
Collection<FileAnnotation> warnings = new IarParser().parse(openFile("issue8823.txt"));
Iterator<FileAnnotation> iterator = warnings.iterator();
       
For loop: iterator.next();
 
return  iterator.next();
```